client/store/singleUser.js

```diff
@@ -65,18 +75,34 @@ export const fetchUser = (userId) => async (dispatch) => {
 
 export const addToCart =
   (product, quantity = 1) => async (dispatch, getState) => {
-    dispatch(_addToCart(product, quantity));
+
     try {
       const token = window.localStorage.getItem("token");
-      const userId = getState().user.id;
+      const user = getState().user;
+      let cartQuantity;
+      let method = 'POST';
+
+      for(let i = 0; i < user.cart.length; i++) {
+          if(user.cart[i].id === product.id) {
+            method = 'PUT'
+            // user.cart[i].quantity++
+            cartQuantity = user.cart[i].quantity + 1
+            dispatch(updateCartQuantity(product.id, cartQuantity))
```

Comment:
Please call dispatch outside of the loop so it doesn't inadvertently get called multiple times

Reply:
done